### PR TITLE
let partners edit scripts they own which have pilot experiments

### PR DESCRIPTION
**Pull request by davidsbailey**
# Description

Fixes https://codedotorg.atlassian.net/browse/LP-1080, which contains some explanation of the problem.

## Testing story

Manually verified that when a script has an editor experiment and a pilot experiment, that a user in that editor experiment (but not the pilot experiment) can view and edit that script. Also added unit tests.

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/config/scripts/time4cs-experiment-unit-2.script [hunk 1]
```diff
@@ -1,5 +1,6 @@
 login_required true
 hideable_stages true
+pilot_experiment 'time4cs-experiment'
```
Comment (dmcavoy) [inline dashboard/config/scripts/time4cs-experiment-unit-2.script:3]: I feel like I remember broward having 2 scripts. Does the other one need to be updated?

**Review by dmcavoy — APPROVED** [review]
Looks good. Just the one question about Broward having 2 scripts

File: dashboard/config/scripts/time4cs-experiment-unit-2.script (hunk 1 above)
Comment (davidsbailey) [reply dashboard/config/scripts/time4cs-experiment-unit-2.script:3]: Hey Dani, they actually have 4, but the other 3 were already marked with this pilot for some reason.

Comment (dmcavoy) [reply dashboard/config/scripts/time4cs-experiment-unit-2.script:3]: Ah ok!

**Pull request merged**